src/default-directory-provider.coffee

```diff
@@ -0,0 +1,42 @@
+{Directory} = require 'pathwatcher'
+fs = require 'fs-plus'
+path = require 'path'
+
+module.exports =
+class DefaultDirectoryProvider
+
+  # Public: Create a Directory that corresponds to the specified URI.
+  #
+  # * `uri` {String} The path to the directory to add.
+  # * `existingPaths` {Array} The existing paths in atom.project.
+  # Returns:
+  # * {Directory} if the given URI is compatible with this provider.
+  # * {string} if the uri maps to a path an existingPath, in which case
+  #     the path in existingPaths is returned.
```

Comment:
Yeah, good point. If not, I'd like to implement said behavior so that we can simplify the `DirectoryProvider` interface.